frontend/src/lib/components/LemonTable/TableRow.tsx

```diff
@@ -36,8 +38,9 @@ function TableRowRaw<T extends Record<string, any>>({
                             <LemonButton
                                 type={isRowExpanded ? 'highlighted' : 'stealth'}
                                 onClick={() => {
-                                    setIsRowExpanded((state) => !state)
+                                    setIsRowExpanded(!isRowExpanded)
                                     !isRowExpanded && expandable.onRowExpand?.(record)
+                                    isRowExpanded && expandable?.onRowCollapse?.(record)
```

Comment:
At this point this should probably be an `if {} else {}`!

Reply:
very true